build/templates/matchers.py.mako

```diff
@@ -0,0 +1,194 @@
+            print("Unexpected type. Expected: {0}. Received: {1}".format(ctypes.POINTER(self.expected_type), type(other)))
+            return False
+        return True
+
+
+class _BufferMatcher(object):
+    def __init__(self, expected_element_type, expected_size_or_value):
+        if isinstance(expected_size_or_value, int):
+            # Were given the size of the buffer
+            self.expected_value = None
+            self.expected_size = expected_size_or_value
+        else:
+            # Were given a list or something that behaves like a list
+            self.expected_value = expected_size_or_value
+            self.expected_size = len(expected_size_or_value)
+        self.expected_type = expected_element_type * self.expected_size
+
+    def __eq__(self, other):
+        if not isinstance(other, self.expected_type):
+            print("Unexpected type. Expected: {0}. Received: {1}".format(self.expected_type, type(other)))
+            return False
+        if self.expected_size != len(other):
+            print("Unexpected length. Expected: {0}. Received: {1}".format(self.expected_size, len(other)))
+            return False
+        if self.expected_value is not None:
+            # Can't compare the objects directly because they're different types (one is list, another is ctypes array).
+            # Go element by element, which allows for reporting the first index where different values were found.
+            for i in range(0, len(self.expected_value)):
+                if self.expected_value[i] != other[i]:
+                    print("Unexpected value at index {0}. Expected: {1}. Received: {2}".format(i, self.expected_value, self.expected_value))
+                    return False
+        return True
+
+
+# Strings
+
+
+class ViStringMatcher(object):
+    def __init__(self, expected_string_value):
+        self.expected_string_value = expected_string_value
+
+    def __eq__(self, other):
+        if not isinstance(other, ctypes.Array):
+            print("Unexpected type. Expected: {0}. Received: {1}".format(ctypes.Array, type(other)))
+            return False
+        if len(other) < len(self.expected_string_value) + 1:  # +1 for NULL terminating character
+            print("Unexpected length in C string. Expected at least: {0}. Received {1}".format(len(other), len(self.expected_string_value) + 1))
+            return False
+        if not isinstance(other[0], bytes):
+            print("Unexpected type. Not a string. Received: {0}".format(type(other[0])))
+            return False
+        if other.value.decode("ascii") != self.expected_string_value:
+            print("Unexpected value. Expected {0}. Received: {1}".format(self.expected_string_value, other.value.decode))
+            return False
+        return True
+
+
+# Scalars
+
+
+class ViBooleanMatcher(_ScalarMatcher):
+    def __init__(self, expected_value):
+        _ScalarMatcher.__init__(self, visatype.ViBoolean, 1 if expected_value is True else False)
```

Comment:
You compare against 1 if expected_value is True, but compare against False when expected_value is False.

Reply:
Fixed!